pandas/tests/test_testing.py

```diff
@@ -215,6 +215,13 @@ def test_multiindex_dtype(self):
                 {'a':[1.0,2.0],'b':[2.1,1.5],'c':['l1','l2']}, index=['a','b'])
         self._assert_not_equal(df1, df2, check_index_type=True)
 
+    def test_empty_dtypes(self):
```

Comment:
Are you sure?  This is testing `assert_frame_equal`, not anything (explicitly) to do with indexing (only needed because `.iloc` is used inside of `assert_frame_equal`)